haystack/components/generators/chat/openai.py

```diff
@@ -486,12 +498,15 @@ def _convert_streaming_chunks_to_chat_message(
                     _arguments=call_data["arguments"],
                 )
 
+        # finish_reason is in the last chunk if usage is not included, and in the second last chunk if usage is included
+        finish_reason = (chunks[-2] if chunk.usage and len(chunks) >= 2 else chunks[-1]).meta.get("finish_reason")
+
```

Comment:
Another alternative to this is below. 
```
if chunk[].meta.get("usage") is not None and len(chunks) >= 2:
        finish_reason = chunks[-2].meta.get("finish_reason")
else:
        finish_reason = chunk.choices[0].finish_reason
```

Reply:
I was wondering, do you think it's possible to have `chunk.usage = True` and a single chunk in `chunks` (e.g. if model response is like "ok")? If yes, that logic might fail.

Reply:
I couldn't find any scenarios in community discussions and docs where only a usage chunk is returned with no preceding content chunk. But theoretically it might be possible, so below is a suggested update to handle this edge case. WDYT?

```suggestion
        if chunk.usage is not None:
              # If there is only one chunk, default finish_reason to None
              finish_reason = chunks[-2].meta.get("finish_reason") if len(chunks) >= 2 else None
        else:
              finish_reason = chunks[-1].meta.get("finish_reason")


```

Reply:
Right, I agree. We can handle the edge case separately if it occurs.